### ISSUE TITLE
Interpolation-only expressions false positive if variable used as key name in map

### ISSUE THREAD

**Issue by grommir** [terraform-linters/tflint#2246]
### Summary

Removing interpolation will result in an error:
```
╷
│ Error: Ambiguous attribute key
│
│   on eks.tf line 189, in resource "kubernetes_secret" "clickhouse_users":
│  189:     var.clickhouse_user = sha256(random_password.clickhouse_user.result)
│
│ If this expression is intended to be a reference, wrap it in parentheses. If it's instead intended as a literal name containing periods, wrap it in quotes to create a string literal.
```

### Command

tflint

### Terraform Configuration

```terraform
resource "kubernetes_secret" "clickhouse_users" {
  metadata {
    name      = "clickhouse-users"
    namespace = "dev"
  }

  data = {
    "${var.clickhouse_user}" = sha256(random_password.clickhouse_user.result)
  }

  type = "Opaque"

}
```

### TFLint Configuration

```HCL
default
```

### Output

```Shell
Warning: Interpolation-only expressions are deprecated in Terraform v0.12.14 (terraform_deprecated_interpolation)

  on eks.tf line 189:
 189:     "${var.clickhouse_user}" = sha256(random_password.clickhouse_user.result)

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.0/docs/rules/terraform_deprecated_interpolation.md
```

### TFLint Version

0.55.1

### Terraform Version

1.10.5

### Operating System

- [x] Linux
- [ ] macOS
- [ ] Windows

**Comment by bendrucker** [comment]
Duplicate of #1462
